src/nus/cs2103/addressbook/AddressBook.java

```diff
@@ -30,188 +39,258 @@
+     */
+    private static final char INPUT_COMMENT_MARKER = '#';
 
     /*
      * This variable is declared for the whole class (instead of declaring it
      * inside the readUserCommand() method to facilitate automated testing using
      * the I/O redirection technique. If not, only the first line of the input
-     * text file will be processed. TODO FIND OUT WHY
+     * text file will be processed.
      */
     private static final Scanner SCANNER = new Scanner(System.in);
 
-    // Model tracking all persons in the address book
+    /**
+     * List of all persons in the address book.
+     */
     private static final List<String[]> ALL_PERSONS = new ArrayList<>();
 
+
     /**
-     * Stores the last viewed person listing (from commands like find and list)
-     * Commands using an index to target a person will refer to the indices in this list.
-     *
-     * Persons in this list are the same objects as the corresponding elements in the backing model.
-     * Deleting persons in the backing model may not necessarily update this latest view snapshot.
+     * Stores the most recent list of persons shown to the user as a result of a user command.
+     * This is a subset of the full list. Deleting persons in the pull list does not delete
+     * those persons from this list.
      */
     private static List<String[]> latestPersonListingView = getAllPersonsInAddressBook(); // initial view is of all
-    private static String latestUserInput;
 
-    private static String storageFilePath;
+    /**
+     * Latest command entered by the user.
+     */
+    private static String userCommand;
 
-    private static boolean isExitRequested = false;
+    /**
+     * The path to the file used for storing person data.
+     */
+    private static String storageFilePath;
 
     /*
      * ==============NOTE TO STUDENTS======================================
-     * Notice how this method solves the whole problem at a very high level. We
-     * can understand the high-level logic of the program by reading this method
-     * alone.
+     * Notice how this method solves the whole problem at a very high level.
+     * We can understand the high-level logic of the program by reading this
+     * method alone.
      * ====================================================================
      */
     public static void main(String[] args) {
+        showWelcomeMessage();
         processProgramArgs(args);
-        showToUser(MESSAGE_WELCOME);
-        loadAddressBookModelFromStorage();
-        while (!isExitRequested) {
-            latestUserInput = getUserInput();
-            String feedback = parseAndExecuteCommand(latestUserInput);
+        loadDataFromStorage();
+        while (true) {
+            userCommand = getUserInput();
+            echoUserCommand(userCommand);
+            String feedback = executeCommand(userCommand);
             showToUser(feedback);
         }
```

Comment:
Went back to `while(true)` loop because it was not clear from this code where the flag is being set.  This method should be very high level and self-explanatory.